### DISCUSSION TITLE
Unstable delay  when running on GPU T4

### DISCUSSION THREAD

**Discussion by xaiocaibi** [inducer/pycuda#417] [Q&A]
**Describe the bug**
When I use pycuda to execute the compiled trt model, there will be regular delay glitches in the program. When locating the problem, it was found that the location where the delay became high was applying for memory. Our guess is that the machine is performing certain tasks on a regular basis to cause this problem.
When nvidia-smi is executed while stress testing, it is found that the delay is unstable. Can you help me with a question?

The figure below is the phenomenon of time delay：
![捕获](https://github.com/inducer/pycuda/assets/18330731/b41df1bb-6711-4cb5-8de4-53abdffb2de7)
When executing the nvida-smi command：
![nvidia](https://github.com/inducer/pycuda/assets/18330731/4a331754-4b17-4dfe-96ec-d7a6f7b9bfd7)

I'm sure this problem is caused by executing the nvidia-smi command, when I remove nvidia-smi, the delay returns to normal.
But this problem still needs to be solved. Can you help me with a question?
Code with latency fluctuations：
![mem](https://github.com/inducer/pycuda/assets/18330731/6cb6b120-b007-47f7-98ad-208a8b7543bb)

**To Reproduce**

**Expected behavior**

**Environment (please complete the following information):**
 - OS: centos
 - CUDA version:11.4
 - CUDA driver version: [e.g. 470.86]
  - PyCUDA version: 2021.1
  - Python version: 3.7
  - GPU T4

**Additional context**

**Comment by inducer** [comment]
Maybe try using a memory pool? Memory allocations, particularly repeated ones, are quite expensive on GPUs.
